enterprise/backend/src/metabase_enterprise/data_editing/params.clj

```diff
@@ -0,0 +1,207 @@
+(ns metabase-enterprise.data-editing.params
+  (:require
+   [metabase-enterprise.data-editing.data-editing :as data-editing]
+   [metabase.actions.core :as actions]
+   [metabase.api.common :as api]
+   [metabase.util :as u]
+   [toucan2.core :as t2]))
+
+(defn- param-value
+  [param-mapping row-delay]
+  (case (:sourceType param-mapping)
+    "constant" (:value param-mapping)
+    "row-data" (when row-delay (get @row-delay (keyword (:sourceValueTarget param-mapping))))
+    nil))
+
+(defn- field-input-type
+  [field field-values]
+  (case (:type field-values)
+    (:list :auto-list :search) "dropdown"
+    (condp #(isa? %2 %1) (:semantic_type field)
+      :type/Description "textarea"
+      :type/Category    "dropdown"
+      :type/FK          "dropdown"
+      (condp #(isa? %2 %1) (:base_type field)
+        :type/Date     "date"
+        :type/DateTime "datetime"
+        "text"))))
+
+(defn- describe-table-action
+  [& {:keys [action-kw
+             table-id
+             param-mapping
+             dashcard-viz
+             row-delay]}]
+  (let [table-id                    table-id
+        table                       (api/read-check (t2/select-one :model/Table :id table-id :active true))
+        field-name->mapping         (u/index-by :parameterId param-mapping)
+        fields                      (-> (t2/select :model/Field :table_id table-id {:order-by [[:position]]})
+                                        (t2/hydrate :dimensions
+                                                    :has_field_values
+                                                    :values))
+        ;; TODO: this should lives in our configuration
```

Comment:
So you're right this is a bit weird, but we won't be saving this in our configuration either.

The relates to non-saved `data-grid.row/*` actions being called from within an Editable's scope. The logic we want is that when we're calling those primitives (which can't have config) from that scope, we use the configuration of the corresponding Editable to generate the default form.

We'll want to encapsulate this inside a multi-method fro the corresponding actions - basically the same method that will generalize the malli spec method we have right now - something like `(describe-input :data-grid.row/create scope input params)`